src/components/KButton/KButton.vue

```diff
@@ -136,17 +136,17 @@ const buttonType = computed((): string => props.to ? 'router-link' : 'button')
 
 const iconColor = computed((): string => {
   if (props.disabled) {
-    return 'var(--grey-400)'
+    return 'var(--grey-400, var(--kui-color-text-neutral-weak, $kui-color-text-neutral-weak))'
   } else if (['primary', 'danger', 'creation'].includes(props.appearance)) {
     return 'white'
   } else if (props.appearance === 'secondary') {
-    return 'var(--KButtonSecondaryColor, var(--blue-600, color(blue-600)))'
+    return 'var(--KButtonSecondaryColor, var(--blue-600, var(--kui-color-text-primary-stronger, $kui-color-text-primary-stronger)))'
   } else if (props.appearance === 'outline') {
-    return 'var(--KButtonOutlineColor, var(--blue-500, color(blue-500)))'
+    return 'var(--KButtonOutlineColor, var(--blue-500, var(--kui-color-text-primary, $kui-color-text-primary)))'
   } else if (props.appearance === 'btn-link') {
-    return 'var(--KButtonLink, var(--blue-500, color(blue-500)))'
+    return 'var(--KButtonLink, var(--blue-500, var(--kui-color-text-primary, $kui-color-text-primary)))'
   } else if (props.appearance === 'btn-link-danger') {
-    return 'var(--KButtonLinkDanger, var(--red-500, color(red-500)))'
+    return 'var(--KButtonLinkDanger, var(--red-500, var(--kui-color-text-danger, $kui-color-text-danger)))'
```

Comment:
issue: you cannot use SCSS variables in JavaScript. You'd need to import the JavaScript tokens and utilize them as the fallback:

```ts
import { KUI_COLOR_TEXT_PRIMARY_STRONGER } from '@kong/design-tokens/dist/tokens/js'

// and then to use it:
return `var(--KButtonSecondaryColor, var(--blue-600, var(--kui-color-text-primary-stronger, KUI_COLOR_TEXT_PRIMARY_STRONGER)))`
```

Reply:
The old `color(blue-500)` instances above were JS functions, not SCSS